drivers/bus/acpi/buspdo.c

```diff
@@ -1358,10 +1361,19 @@ Bus_PDO_QueryResourceRequirements(
       return STATUS_UNSUCCESSFUL;
     }
 
-    resource= Buffer.Pointer;
+    SeenStartDependent = FALSE;
+    resource = Buffer.Pointer;
     /* Count number of resources */
-    while (resource->Type != ACPI_RESOURCE_TYPE_END_TAG)
+    while (resource->Type != ACPI_RESOURCE_TYPE_END_TAG && resource->Type != ACPI_RESOURCE_TYPE_END_DEPENDENT)
     {
+        if (resource->Type == ACPI_RESOURCE_TYPE_START_DEPENDENT)
+        {
+            if (SeenStartDependent)
+            {
+                break;
```

Comment:
Does this happen?
Is this something we want to log?

Reply:
Yes, it's completely expected. I was going to paste an example in the commit message but that seemed excessive. I added it to the PR description, hope that clears things up.